src/sw/redis++/connection.cpp

```diff
@@ -22,6 +22,13 @@
 #include "command.h"
 #include "command_args.h"
 
+#if defined _WIN32
+struct timeval {
+        long    tv_sec;         /* seconds */
+        long    tv_usec;        /* and microseconds */
+};
+#endif
+
```

Comment:
hiredis' solution for timeval:

```
#ifdef _MSC_VER

#include <winsock2.h> /* for struct timeval */

#endif
```

hiredis' solution seems better. It can avoid name conflict.